plotly_resampler/aggregation/aggregation_interface.py

```diff
@@ -4,169 +4,260 @@
-                    index=df_gap_idx, data=None, name=s.name, copy=False, dtype=s.dtype
-                )
-
-                if isinstance(df_res_gap.index, pd.DatetimeIndex):
-                    # Due to the s.index`.values` cast, df_res_gap has lost
-                    # time-information, so now we restore it
-                    df_res_gap.index = df_res_gap.index.tz_localize("UTC").tz_convert(
-                        s.index.tz
-                    )
-
-                # Note:
-                #  * the order of pd.concat is important for correct visualization
-                #  * we also need a stable algorithm for sorting, i.e., the equal-index
-                #    data-entries their order will be maintained.
-                s = pd.concat([df_res_gap, s], ignore_index=False).sort_index(
-                    kind="mergesort"
-                )
-        return s
-
-    def _replace_gap_end_none(self, s: pd.Series) -> pd.Series:
-        # ------- REPLACE None where a gap ends -------
-        med_diff, s_idx_diff = self._calc_med_diff(s)
-        if med_diff is not None:
-            # Replace data-points with None where the gaps occur
-            # The default is the end of a gap
-            nan_mask = s_idx_diff > 4 * med_diff
-            if self.nan_position == "begin":
-                # Replace the last non-gap datapoint (begin of gap) with Nan
-                nan_mask = np.roll(nan_mask, -1)
-            elif self.nan_position == "both":
-                # Replace the encompassing gap datapoints with Nan
-                nan_mask |= np.roll(nan_mask, -1)
-            s.loc[nan_mask] = None
-        return s
-
-    def aggregate(self, s: pd.Series, n_out: int) -> pd.Series:
-        """Aggregate (downsample) the given input series to the given n_out samples.
+        Parameters
+        ----------
+        x : np.ndarray, optional
+            The x-axis data, by default None
+        y : np.ndarray
+            The y-axis data
+        n_out : int, optional
+            The number of output datapoints, by default None
+        kwargs : dict
+            Additional keyword arguments that are passed to the `_aggregate` method.
+
+        Returns
+        -------
+        Tuple[np.ndarray, np.ndarray]
+            The aggregated x and y data, respectively
+
+        """
+        assert n_out is not None
+
+        # assert isinstance(y, np.ndarray)
+        assert y.ndim == 1
+        DataAggregator._supports_dtype(y, self.y_dtype_regex_list)
+
+        if x is not None:
+            assert x.ndim == 1
+            assert x.shape == y.shape
+            DataAggregator._supports_dtype(x, self.x_dtype_regex_list)
+
+        return self._aggregate(x, y, n_out=n_out, **kwargs)
+
+
+class DataPointSelector(AbstractAggregator, ABC):
+    """Implementation of the AbstractAggregator interface for data point selection.
+
+    A data point selector is an aggregator that selects data points, and thus doesn't
+    aggregate the data.
```

Comment:
This seems pretty confusing from a documentation perspective. We're basically repeating the class definition twice here.
What does it mean to select? How is it different to aggregation? And also most importantly - when and how should this be used.